model/task_lifecycle.go

```diff
@@ -1570,24 +1570,36 @@ func resetStrandedTask(t *task.Task) error {
 		// If the task has already exceeded the unschedulable threshold, we
 		// don't want to restart it, so just mark it as finished.
 		if t.DisplayOnly {
-			for _, etID := range t.ExecutionTasks {
-				var execTask *task.Task
-				execTask, err := task.FindOneId(etID)
-				if err != nil {
+			var execTasks []task.Task
+			var err error
+
+			if t.ResetFailedWhenFinished && !t.ResetWhenFinished {
```

Comment:
Are you only querying failed tasks in case not all tasks were restarted? We actually should be clearing this field when we do restarts (cc @bynn do we do that?), otherwise we'll be trying to restart on a loop. I think we should just query all execution tasks and mark end any that's not already finished.

Reply:
I changed to query all tasks and then only mark end for those completed.